tools/CoqMakefile.in

```diff
@@ -602,6 +607,18 @@ $(ML4FILES:.ml4=.cmx): %.cmx: %.ml4
 	$(SHOW)'CAMLOPT -pp -c $(FOR_PACK) $<'
 	$(HIDE)$(CAMLOPTC) $(CAMLDEBUG) $(CAMLFLAGS) $(CAMLPKGS) $(PP) $(FOR_PACK) -impl $<
 
+$(MLGFILES:.mlg=.ml): %.ml: %.mlg
+	$(SHOW)'COQPP $<'
+	$(HIDE)$(COQPP) $<
+
+$(MLGFILES:.mlg=.cmo): %.cmo: %.ml
```

Comment:
This part can be factored with the regular ml files if I'm not mistaken, eg `$(MLFILES:.ml=.cmo) $(MLGFILES:.mlg=.cmo): %.cmo: %.ml`

Reply:
This syntax doesn't seem to work.

Reply:
maybe you have a ALLCMO files variable already containing that union...

Reply:
I don't like expanding the set of global variables, I think I prefer the current hackish but local solution.